components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/message/OAuthMessage.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    public SessionDataCacheEntry getResultFromLogin() {
+        return resultFromLogin;
+    }
+
+    public void setResultFromLogin(SessionDataCacheEntry resultFromLogin) {
+        this.resultFromLogin = resultFromLogin;
+    }
+
+    public SessionDataCacheEntry getResultFromConsent() {
+        return resultFromConsent;
+    }
+
+    public void setResultFromConsent(SessionDataCacheEntry resultFromConsent) {
+        this.resultFromConsent = resultFromConsent;
+    }
+
+    public boolean isForceAuthenticate() {
+        return forceAuthenticate;
+    }
+
+    public void setForceAuthenticate(boolean forceAuthenticate) {
+        this.forceAuthenticate = forceAuthenticate;
+    }
+
+    public boolean isPassiveAuthentication() {
+        return isPassiveAuthentication;
+    }
+
+    public void setPassiveAuthentication(boolean passiveAuthentication) {
+        this.isPassiveAuthentication = passiveAuthentication;
+    }
+
+    public SessionDataCacheEntry getSessionDataCacheEntry() {
+        return sessionDataCacheEntry;
+    }
+
+    public void setSessionDataCacheEntry(SessionDataCacheEntry sessionDataCacheEntry) {
+        this.sessionDataCacheEntry = sessionDataCacheEntry;
+    }
+
+    /**
+     * In federated and multi steps scenario there is a redirection from commonauth to samlsso so have to get
+     * session data key from query parameter
+     *
+     * @param req Http servlet request
+     * @return Session data key
+     */
+    private String getSessionDataKey(HttpServletRequest req) {
+        String sessionDataKey = (String) req.getAttribute(OAuthConstants.SESSION_DATA_KEY);
+        if (sessionDataKey == null) {
+            sessionDataKey = req.getParameter(OAuthConstants.SESSION_DATA_KEY);
+        }
+        return sessionDataKey;
+    }
+
+    public boolean isConsentResponseFromUser() {
+        return resultFromConsent != null;
+    }
+
+    public boolean isAuthResponseFromFramework() {
+        return resultFromLogin != null;
+    }
+
+    public boolean isInitialRequest() {
+        return request.getParameter("client_id") != null && getSessionDataKey(request) == null
+                && request.getParameter(OAuthConstants.SESSION_DATA_KEY_CONSENT) == null;
+    }
+
+    public Object getFlowStatus() {
+        return request.getAttribute(FrameworkConstants.RequestParams.FLOW_STATUS);
+    }
+
+    public boolean isRequestToCommonauth() {
+        return Boolean.parseBoolean(request.getParameter(FrameworkConstants.RequestParams.TO_COMMONAUTH));
+    }
+
+    public String getOauthPKCECodeChallenge() {
+        return request.getParameter(OAuthConstants.OAUTH_PKCE_CODE_CHALLENGE);
+    }
+
+    public String getOauthPKCECodeChallengeMethod() {
+        return request.getParameter(OAuthConstants.OAUTH_PKCE_CODE_CHALLENGE_METHOD);
+    }
+
+    @Override
+    public String toString() {
+        return "OAuthMessage{" +
+                "properties=" + properties +
```

Comment:
Will concatenating the map provide the desired output?

Reply:
In most cases, Do we have an alternative way?